### PR TITLE
Fix API usage in formspecs.

### LINKED ISSUE #2: `formspecs` mod uses `minetest.get_password_hash` to generate a random value.
This is a misuse of the API (see https://github.com/minetest/minetest/issues/12564). 

The call can easily be replaced with `SecureRandom()` with a fall back to `PcgRandom()` if needed.

https://github.com/mt-historical/blockcolor/blob/c9673dea5d9e23f5a0e406bc85c19600362ded63/mods/Other/formspecs/init.lua#L212

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by wsor4035** [comment]
> The constructor returns nil if a secure random device cannot be found on the system.

in practicality does this actually happen on any platform? windows/linux/android/macos?

**Comment by red-001** [comment]
Not sure this security check is even needed that much now that minetest checks formspecs itself, but that old check wasn't very strong.

`session_seed` was brute-forceble.

**Comment by red-001** [comment]
> > The constructor returns nil if a secure random device cannot be found on the system.
> 
> in practicality does this actually happen on any platform? windows/linux/android/macos?

Probably not, and if it did the server would likely be unable to handle logins anyways since it needs entropy for SRP.

**Comment by wsor4035** [comment]
the whole thing isnt very good, given it writes into the minetest global space.

anyways, thanks

**Comment by red-001** [comment]
Flagged the API documentation weirdness with the project #14991

File: mods/Other/formspecs/init.lua [hunk 1]
```diff
@@ -14,12 +14,21 @@ minetest.FORMSPEC_SIGTERM = 3		-- server is shutting down
 minetest.FORMSPEC_SIGPROC = 4		-- procedural closure
 minetest.FORMSPEC_SIGTIME = 5		-- timeout reached
 
+local srng = SecureRandom()
+if srng ~= nil then
+	minetest.log("error", "Unable to initalize random number generator, formspecs may be insecure.")
```
Comment (rudzik8) [inline mods/Other/formspecs/init.lua:19]: Shouldn't the check be `srng == nil`? Like, isn't the error all about `SecureRandom()` returning `nil`?

Comment (red-001) [reply mods/Other/formspecs/init.lua:19]: Yeah it should be, my bad, I didn't check what i was writing well enough.

Comment (wsor4035) [reply mods/Other/formspecs/init.lua:19]: probably should be `if not srng`. worst case its just nagging now

**Pull request merged**